client-course-schedulizer/src/readCSV.ts

```diff
@@ -0,0 +1,428 @@
+import papa from "papaparse";
+import * as di from "./dataInterfaces";
+
+// For now the plan is to assume the CSV has these proper headers
+const validFields = [
+  "name",
+  // If both are found, use "prefixes"
+  "prefix",
+  "prefixes",
+  "number",
+  // Corresponds to letter
+  "section",
+  "studentHours",
+  "facultyHours",
+  // If both are found, use "startTimeStr"
+  "startTime",
+  "startTimeStr",
+  "duration",
+  // Covers both building and room number
+  "location",
+  "roomCapacity",
+  "year",
+  "term",
+  "half",
+  // Needs to be parsed
+  "days",
+  "globalMax",
+  "localMax",
+  "anticipatedSize",
+  // If both are found, use "instructors"
+  "instructor",
+  "instructors",
+  "comments",
+];
+
+// Define regexes for parsing
+const timeReg = RegExp("(?<![1-9])(1[0-9]|2[0-3]|[0-9]):([0-5][0-9])");
+const amReg = RegExp("[Aa][Mm]");
+const pmReg = RegExp("[Pp][Mm]");
+const fallReg = RegExp("[Ff]");
+const summerReg = RegExp("[Ss][Uu]");
+const springReg = RegExp("[Ss]");
+// "W" represents interim in Pruim's system it seems
+const interimReg = RegExp("[Ii]|[Ww]");
+const firstReg = RegExp("[Ff]irst");
+const secondReg = RegExp("[Ss]econd");
+const fullReg = RegExp("[Ff]ull");
+const sunReg = RegExp("[Ss][Uu]|[Nn]");
+const monReg = RegExp("[Mm]");
+const tuesReg = RegExp("[Tt](?![Hh])");
+const wedReg = RegExp("[Ww]");
+const thursReg = RegExp("[Tt][Hh]|[Rr]");
+const friReg = RegExp("[Ff]");
+const satReg = RegExp("[Ss](?![Uu])");
```

Comment:
I think the time regex is great and the others are helpful, maybe we should convert all of the text to uppercase or lowercase to ease the complexity? Since you've written it out already, I think this is probably fine to leave it as is.

Reply:
In hindsight, this probably would have been a good idea, but I agree that since it's already all there, it might just be best to leave it as is for now.